### PR TITLE
include Authorization on environment.rb and api code cleanup 

### PR DESCRIPTION
**Pull request by isratrade**

### CODE REVIEW

File: app/controllers/api/v1/architectures_controller.rb [hunk 1]
```diff
@@ -1,15 +1,13 @@
...
-      include Foreman::Controller::AutoCompleteSearch
       before_filter :find_resource, :only => %w{show update destroy}
 
       api :GET, "/architectures/", "List all architectures."
       param :search, String, :desc => "filter results"
       param :order,  String, :desc => "sort results"
       def index
-        @architectures = Architecture.search_for(params[:search], :order => params[:order]).
-            paginate(:page => params[:page], :include => :operatingsystems)
+        @architectures = Architecture.includes(:operatingsystems).search_for(params[:search], :order => params[:order]).paginate(:page => params[:page])
```
Comment (ohadlevy) [inline app/controllers/api/v1/architectures_controller.rb]: i think we can drop pagination as well, at least until we make a decision if we want to include it or not

File: app/controllers/architectures_controller.rb [hunk 2]
```diff
@@ -16,6 +16,7 @@ def new
 
   def show
     respond_to do |format|
+      format.html { invalid_request }
```
Comment (ohadlevy) [inline app/controllers/architectures_controller.rb]: if its required, I think you want to generalize it

**Comment by isratrade** [comment]
{ invalid_request } was in the environments_controller, so I included it here.  However, on second thought, why should the show method return nothing or worse, respond an error message for invalid request.

File: app/controllers/architectures_controller.rb [hunk 3]
```diff
@@ -3,10 +3,10 @@ class ArchitecturesController < ApplicationController
   before_filter :find_by_name, :only => %w{show edit update destroy}
 
   def index
-    values = Architecture.search_for(params[:search], :order => params[:order])
+    values = Architecture.includes(:operatingsystems).search_for(params[:search], :order => params[:order])
```
Comment (ohadlevy) [inline app/controllers/architectures_controller.rb]: the reason of putting the include in the html part, was thats an optimization only needed for the html view.

**Comment by ohadlevy** [comment]
{ invalid_request } is there because there is no show html view.

File: app/controllers/api/v1/architectures_controller.rb [hunk 4]
```diff
@@ -1,15 +1,13 @@
...
-      include Foreman::Controller::AutoCompleteSearch
       before_filter :find_resource, :only => %w{show update destroy}
 
       api :GET, "/architectures/", "List all architectures."
       param :search, String, :desc => "filter results"
       param :order,  String, :desc => "sort results"
       def index
-        @architectures = Architecture.search_for(params[:search], :order => params[:order]).
-            paginate(:page => params[:page], :include => :operatingsystems)
+        @architectures = Architecture.includes(:operatingsystems).search_for(params[:search], :order => params[:order])
```
Comment (ohadlevy) [inline app/controllers/api/v1/architectures_controller.rb:10]: if we don't show the os as a nested object in the json output, there is no reason why we should eager load (include) the operatingsystems.

**Comment by isratrade** [comment]
pull request is outdated and can be closed.

**Pull request closed without merging**